File: Algorithm-Swift/Algorithm-Swift/Programmers/수학/멀쩡한사각형.swift
import Foundation

func euclidean(_ big: Int, _ small: Int) -> Int {
    if big % small == 0 {
        return small
    } else {
        return euclidean(small, big%small)
    }
}

func solution(_ w:Int, _ h:Int) -> Int64{
    var width = w
    var height = h
    if w == h { return Int64((w*h)-w) }
    if h > w { (width, height) = (height, width) }
    
    return Int64((width*height) - (w+h-euclidean(width,height)))
}

File: By-Python/programmers/Summer,Winter-Coding(2019)/멀쩡한사각형.py
# 멀쩡한 사각형

def solution(w, h):
    if w == h: return w * h - w

    if w < h:
        w, h = h, w

    def gcd(w, h):
        if h == 0:
            return w
        else:
            return gcd(h, w % h)

    return w * h - (w + h - gcd(w, h))